stdlib/public/core/Mirror.swift

```diff
@@ -905,6 +905,43 @@ extension DictionaryLiteral : RandomAccessCollection {
   }
 }
 
+extension DictionaryLiteral : Equatable where Key: Equatable, Value : Equatable {
+  @_inlineable // FIXME(sil-serialize-all)
+  public static func ==(lhs: DictionaryLiteral<Key, Value>, rhs: DictionaryLiteral<Key, Value>) -> Bool {
+    let lhsCount = lhs.count
+    if lhsCount != rhs.count {
+        return false
+    }
+    // We know that lhs.count == rhs.count, compare element wise.
+    for idx in 0..<lhsCount {
+        if lhs[idx].key != rhs[idx].key ||
+            lhs[idx].value != rhs[idx].value {
+            return false
+        }
+    }
+    return true
+  }
+}
+
+extension DictionaryLiteral : Hashable where Key: Hashable, Value : Hashable {
+  /// The hash value for the dictionary.
```

Comment:
I would avoid calling `self` a "dictionary" here (and in the paragraph below). `DictionaryLiteral` values aren't dictionaries, and, frustratingly, they also aren't dictionary literals. Here is one way to sidestep the naming issue:

> The hash value for the collection.
> 
> Two `DictionaryLiteral` values that are equal will always have equal hash values.
>
> Hash values are not guaranteed etc.